From 3dd8576fc70ed42becb6ec748e5ebee38f275d25 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Fri, 17 Jul 2026 11:20:22 +0200
Subject: [PATCH] segtree: rename set_elem_add() to set_elem_expr_add()

JIRA: https://issues.redhat.com/browse/RHEL-190549
Upstream Status: nftables commit f598264de4bb8f8366e9d93ee2f05bd31d020728

commit f598264de4bb8f8366e9d93ee2f05bd31d020728
Author: Pablo Neira Ayuso <pablo@netfilter.org>
Date:   Thu Feb 5 03:41:15 2026 +0100

    segtree: rename set_elem_add() to set_elem_expr_add()

    Just a clean up, to prepare for the introduction of struct set_elem
    that will provide a set_elem_add() function again.

    Rename it now to leave room for such future change.

    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>
    Signed-off-by: Florian Westphal <fw@strlen.de>

Signed-off-by: Phil Sutter <psutter@redhat.com>
---
 src/segtree.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/segtree.c b/src/segtree.c
index 5924460..3cdecfc 100644
--- a/src/segtree.c
+++ b/src/segtree.c
@@ -53,8 +53,8 @@ static void interval_expr_copy(struct expr *dst, struct expr *src)
 	list_splice_init(&src->stmt_list, &dst->stmt_list);
 }
 
-static void set_elem_add(const struct set *set, struct expr *init, mpz_t value,
-			 uint32_t flags, enum byteorder byteorder)
+static void set_elem_expr_add(const struct set *set, struct expr *init,
+			      mpz_t value, uint32_t flags, enum byteorder byteorder)
 {
 	struct expr *expr;
 
@@ -82,8 +82,8 @@ struct expr *get_set_intervals(const struct set *set, const struct expr *init)
 	list_for_each_entry(i, &init->expressions, list) {
 		switch (i->key->etype) {
 		case EXPR_VALUE:
-			set_elem_add(set, new_init, i->key->value,
-				     i->flags, byteorder);
+			set_elem_expr_add(set, new_init, i->key->value,
+					  i->flags, byteorder);
 			break;
 		case EXPR_CONCAT:
 			compound_expr_add(new_init, expr_clone(i));
@@ -95,11 +95,11 @@ struct expr *get_set_intervals(const struct set *set, const struct expr *init)
 			break;
 		default:
 			range_expr_value_low(low, i);
-			set_elem_add(set, new_init, low, 0, i->byteorder);
+			set_elem_expr_add(set, new_init, low, 0, i->byteorder);
 			range_expr_value_high(high, i);
 			mpz_add_ui(high, high, 1);
-			set_elem_add(set, new_init, high,
-				     EXPR_F_INTERVAL_END, i->byteorder);
+			set_elem_expr_add(set, new_init, high,
+					  EXPR_F_INTERVAL_END, i->byteorder);
 			break;
 		}
 	}
